python_modules/libraries/dagster-aws/dagster_aws/emr/emr_eks_pyspark_step_launcher.py

```diff
@@ -0,0 +1,574 @@
+    DagsterEvent,
+    DagsterEventType,
+    EngineEventData,
+    log_step_event,
+)
+from dagster.core.execution.context.system import StepExecutionContext
+from dagster.core.execution.plan.external_step import step_context_to_step_run_ref
+from dagster.core.execution.plan.objects import (
+    ErrorSource,
+    StepFailureData,
+    UserFailureData,
+)
+from dagster_pyspark.resources import PySparkResource
+
+from dagster_aws.emr.pyspark_step_launcher import (
+    CODE_ZIP_NAME,
+    EMR_PY_SPARK_STEP_LAUNCHER_BASE_CONFIG,
+    EmrPySparkStepLauncherBase,
+)
+
+from . import emr_eks_step_main
+from .monitor import EmrEksJobError, EmrEksJobRunMonitor
+
+EMR_EKS_CONFIG_SCHEMA = dict(
+    **EMR_PY_SPARK_STEP_LAUNCHER_BASE_CONFIG,
+    **{
+        "emr_release_label": Field(StringSource, description="EMR release label"),
+        "container_image": Field(StringSource, description="Container image to use"),
+        "job_role_arn": Field(StringSource, description="ARN of the role to use for job execution"),
+        "log_group_name": Field(StringSource, description="Cloudwatch log group"),
+        "additional_relative_paths": Field(
+            Array(StringSource),
+            description=(
+                "List of relative relative paths which are not correctly reachable from the "
+                "repository base. In practice let's say we have a package located at "
+                "`/root/some/involved/path/target_package` it can still be imported "
+                "as `target_package` if `some/involved/path/` is added to the list. "
+                "This can be useful, for example, in case of git submodules."
+            ),
+            default_value=[],
+        ),
+        "log4j_conf": Field(Permissive(), default_value={}, is_required=False),
+    },
+)
+
+
+@resource(config_schema=EMR_EKS_CONFIG_SCHEMA)
+def emr_eks_pyspark_resource(context):
+    spark_config = context.resource_config.get("spark_config")
+
+    return EmrEksPySparkResource(
+        cluster_id=context.resource_config.get("cluster_id"),
+        container_image=context.resource_config.get("container_image"),
+        deploy_local_job_package=context.resource_config.get("deploy_local_job_package"),
+        emr_release_label=context.resource_config.get("emr_release_label"),
+        job_role_arn=context.resource_config.get("job_role_arn"),
+        log_group_name=context.resource_config.get("log_group_name"),
+        log4j_conf=context.resource_config.get("log4j_conf"),
+        region_name=context.resource_config.get("region_name"),
+        staging_bucket=context.resource_config.get("staging_bucket"),
+        staging_prefix=context.resource_config.get("staging_prefix"),
+        spark_config=spark_config,
+        ephemeral_instance=context.instance.is_ephemeral,
+        additional_relative_paths=context.resource_config.get("additional_relative_paths"),
+    )
+
+
+class EmrEksPySparkResource(PySparkResource, EmrPySparkStepLauncherBase):
```

Comment:
nit: we should uniformize with the original EMR one -- it is named `EmrPySparkStepLauncher`, so we should have a `EmrEksPySparkStepLauncher`

we also have a resource here inheriting from `PySparkResource`, is it not the case of the original EMR one?

Reply:
As we discussed, it looks like our solution is more correct